code/datums/components/harvester.dm

```diff
@@ -180,6 +180,10 @@
 	switch(loaded_reagent)
 		if(/datum/reagent/medicine/tramadol)
 			target.apply_damage(weapon.force*0.6, BRUTE, user.zone_selected)
+			if(isxeno(target))
+				var/mob/living/carbon/xenomorph/xeno_target = target
+				if(xeno_target.endure)
+					return
```

Comment:
Using return here will prevent this proc from reaching certain sections after the switch case, such as informing the user if the reagent storage is empty. I recommend just throwing the status application into the inverted if conditions instead.

Reply:
return is used later in the code in bicardine section, does it cause issues?

Reply:
For bicardine it is correct since you do not want the reagent effect to fire at all when attacking xenos since it heals.
Here however, you still apply the bonus AP damage, which basically means on enduring Ravagers you'd get free bonus damage on every stab (since it'd not consume the charge)